V3 compatible, Recalc improved, progress bars fixed

mortii opened the pull request:
- Now works on the V3 scheduler (fixes #276)
- Exponentially faster recalc action
- Progress dialogs are now shown when recalcing and anki no longer freezes (background thread is used instead of main thread)
- Renamed some files that were poorly named
- Refactored init file and extracted hidden hooks to make the code execution logic much clearer
- Added some type hints for clearer logic
- refactored some variable names to follow PEP conventions

landonepps reviewed (changes requested):
Overall looks good, I appreciate the attention to detail like renaming variables.
I'm happy to merge it if you can address the linter warnings.

Also, the `master` branch is outdated. The latest version on AnkiWeb is using the `qt6-migration` branch. I'll merge your changes into that branch after it gets merged to master.

mortii commented on morph/main.py:
```diff
@@ -149,15 +139,15 @@ def mkAllDb(all_db=None):
-        if i % 500 == 0:
-            mw.progress.update(value=i)
+        # if i % 500 == 0:
+        mw.taskman.run_on_main(lambda: mw.progress.update(value=i))
```
aah, nice!

mortii committed:
V3 compatible, Recalc improved, progress bars fixed

- Now works on the V3 scheduler
- Exponentially faster recalc action
- Progress dialogs are now shown when recalcing and anki no longer freezes (background thread is used instead of main thread)
- Renamed some files that were poorly named
- Refactored init file and extracted hidden hooks to make the code execution logic much clearer
- Added some type hints for clearer logic
- refactored some variable names to follow PEP conventions

mortii committed:
Updated development section of README

mortii commented:
I fixed the lint errors and I rebased onto qt6-migration

mortii commented:
hold on, a very a sneaky error appeared that needs to be fixed

mortii commented:
there was an invisible input parameter after removing a lambda, but it's fixed now.

mortii commented:
Also this should probably be placed in the ankiweb description until we get a proper wiki/guide (next on my todo-lis):

> NB! If the deck you are studying has sub-decks then new cards will by default only be gathered from the first sub-deck until it is empty before looking for new cards in the next sub-deck. If you instead want to get new i+1 cards from all sub-decks do the following:
> 1. Activate the v3 scheduler: Tools -> Review -> Scheduler -> V3 scheduler
> 2. Deck that has sub-decks: Deck options -> Display Order -> New card gather order -> Ascending position

The writing is kinda clunky, feel free to improve it in whatever way you see fit , lol.

landonepps commented:
Looks good to me, thanks for all the work you did here.
And I appreciate the AnkiWeb note, I'll be sure to add it.

PrentissLiu commented:
Thank you so much for your work. I am the version of QT5. I tried to modify: 
readability_ui.py and readability_settings_ui.py : 

> from PyQt6 import QtCore, QtGui, QtWidgets

to:

> from PyQt5 import QtCore, QtGui, QtWidgets

it works!

landonepps commented:
Awesome, I really appreciate the report!
Both of my computers are arm-based Macs, which don't support Qt5, making it quite annoying to test.

If this is really all that's needed, I should be able to modify the UI generation code to replace these imports with the aqt version and have full backwards compatibility.
